hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/ConsumerMessageSenderFactory.java

```diff
@@ -74,9 +76,13 @@ public ConsumerMessageSender create(Subscription subscription, ConsumerRateLimit
                 futureAsyncTimeout);
     }
 
-    private ExecutorService createRateLimiterReportingExecutor(int size) {
-        ExecutorService wrapped = Executors.newFixedThreadPool(size);
-        return new InstrumentedExecutorService(wrapped, hermesMetrics, "rate-limiter-reporter");
+    private ExecutorService createRateLimiterReportingExecutor(int size, boolean monitoringEnabled) {
```

Comment:
we repeat same 'if' statement 3 times, maybe it's time to create wrapper/factory?